dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/dao/SimpleMongoDBStorageStrategy.java

```diff
@@ -85,14 +89,14 @@ public DBObject getQuery(final RyaStatement stmt) {
         final RyaURI context = stmt.getContext();
         final BasicDBObject query = new BasicDBObject();
         if (subject != null){
-            query.append(SUBJECT, subject.getData());
+            query.append(SUBJECT_HASH, DigestUtils.sha256Hex(subject.getData()));
```

Comment:
I can store as either.  Not really sure if there are any pros-cons between the two